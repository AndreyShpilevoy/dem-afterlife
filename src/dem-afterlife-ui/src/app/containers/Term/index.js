/* eslint fp/no-class: 0, fp/no-nil: 0, fp/no-unused-expression: 0, fp/no-mutation: 0, fp/no-this: 0 */

import React, {PureComponent} from 'react';
import {shape, number, string, bool} from 'prop-types';
import {connect} from 'react-redux';
import {localization, defaults} from 'utils';

class Term extends PureComponent {
    static propTypes = {
        term: shape({
            id: number.isRequired,
            value: string.isRequired
        }),
        className: string,
        spaceBefore: bool,
        spaceAfter: bool,
        indentBefore: bool,
        indentAfter: bool,
        untermedPostfix: string,
        locale: string.isRequired,
        doNotTerm: bool
    };

    static defaultProps = {
        untermedPostfix: defaults.emptyString,
        doNotTerm: false
    };

    render() {
        const {term, locale, className, spaceBefore, spaceAfter, indentBefore, indentAfter, untermedPostfix, doNotTerm} = this.props;
        const spaceBeforeString = spaceBefore ? defaults.spaceString : defaults.emptyString;
        const spaceAfterString = spaceAfter ? defaults.spaceString : defaults.emptyString;
        const termedString = doNotTerm ? defaults.emptyString : localization.getTermTranslation(term, locale);
        const content = `${spaceBeforeString}${termedString}${untermedPostfix}${spaceAfterString}`;
        const style = {
            paddingLeft: indentBefore ? defaults.spaceWidth : defaults.emptyString,
            paddingRight: indentAfter ? defaults.spaceWidth : defaults.emptyString
        };
        return (
            <span className={className} style={style}>
                {content}
            </span>
        );
    }
}

const mapStateToProps = ({layoutReducer}) => ({
    locale: layoutReducer.locale
});

export default connect(mapStateToProps)(Term);
