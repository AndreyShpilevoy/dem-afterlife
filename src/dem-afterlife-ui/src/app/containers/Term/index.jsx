/* eslint fp/no-class: 0, fp/no-nil: 0, fp/no-unused-expression: 0, fp/no-mutation: 0, fp/no-this: 0 */

import React, {PureComponent} from 'react';
import {shape, number, string, bool} from 'prop-types';
import {connect} from 'react-redux';
import {localization} from 'utils';
import {localeSelector} from 'containers/selectors';

class Term extends PureComponent {
    static propTypes = {
        term: shape({
            id: number.isRequired,
            value: string.isRequired
        }),
        className: string,
        spaceBefore: bool,
        spaceAfter: bool,
        indentBefore: bool,
        indentAfter: bool,
        untermedPostfix: string,
        locale: string.isRequired,
        doNotTerm: bool
    };

    static defaultProps = {
        term: null,
        untermedPostfix: '',
        doNotTerm: false,
        className: '',
        spaceBefore: false,
        spaceAfter: false,
        indentBefore: false,
        indentAfter: false
    };

    render() {
        const {term, locale, className, spaceBefore, spaceAfter, indentBefore, indentAfter, untermedPostfix, doNotTerm} = this.props;
        const spaceBeforeString = spaceBefore ? ' ' : '';
        const spaceAfterString = spaceAfter ? ' ' : '';
        const termedString = doNotTerm ? '' : localization.getTermTranslation(term, locale);
        const content = `${spaceBeforeString}${termedString}${untermedPostfix}${spaceAfterString}`;
        const style = {
            paddingLeft: indentBefore ? '0.25rem' : '',
            paddingRight: indentAfter ? '0.25rem' : ''
        };
        return (
            <span className={className} style={style}>
                {content}
            </span>
        );
    }
}

const mapStateToProps = state => ({
    locale: localeSelector(state)
});

export default connect(mapStateToProps)(Term);
