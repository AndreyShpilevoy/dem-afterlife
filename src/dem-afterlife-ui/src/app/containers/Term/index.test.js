/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0, fp/no-mutation:0 , react/jsx-filename-extension:0 */

import React from 'react';
import {mount} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import Term from './index';

describe('Term', () => {
    const mockStore = configureMockStore();

    it('component match expected snapshot for simple term', () => {
        const props = {
            store: mockStore({sharedReducer: {locale: 'ru'} }),
            term: {id: 1, value: 'Тем'}
        };
        expect(mount(<Term {...props} />, {lifecycleExperimental: true})).toMatchSnapshot();
    });

    it('component match expected snapshot for simple term and untermedPostfix', () => {
        const props = {
            store: mockStore({sharedReducer: {locale: 'ru'} }),
            term: {id: 1, value: 'Тем'},
            untermedPostfix: ':'
        };
        expect(mount(<Term {...props} />, {lifecycleExperimental: true})).toMatchSnapshot();
    });

    it('component match expected snapshot for not term and untermedPostfix', () => {
        const props = {
            store: mockStore({sharedReducer: {locale: 'ru'} }),
            doNotTerm: true,
            untermedPostfix: ':'
        };
        expect(mount(<Term {...props} />, {lifecycleExperimental: true})).toMatchSnapshot();
    });

    it('component match expected snapshot for term with space before', () => {
        const props = {
            store: mockStore({sharedReducer: {locale: 'ru'} }),
            term: {id: 1, value: 'Тем'},
            spaceBefore: true
        };
        expect(mount(<Term {...props} />, {lifecycleExperimental: true})).toMatchSnapshot();
    });

    it('component match expected snapshot for simple term with space after', () => {
        const props = {
            store: mockStore({sharedReducer: {locale: 'ru'} }),
            term: {id: 1, value: 'Тем'},
            spaceAfter: true
        };
        expect(mount(<Term {...props} />, {lifecycleExperimental: true})).toMatchSnapshot();
    });

    it('component match expected snapshot for simple term with space before and after', () => {
        const props = {
            store: mockStore({sharedReducer: {locale: 'ru'} }),
            term: {id: 1, value: 'Тем'},
            spaceBefore: true,
            spaceAfter: true
        };
        expect(mount(<Term {...props} />, {lifecycleExperimental: true})).toMatchSnapshot();
    });

    it('component match expected snapshot for term with indent before', () => {
        const props = {
            store: mockStore({sharedReducer: {locale: 'ru'} }),
            term: {id: 1, value: 'Тем'},
            indentBefore: true
        };
        expect(mount(<Term {...props} />, {lifecycleExperimental: true})).toMatchSnapshot();
    });

    it('component match expected snapshot for simple term with indent after', () => {
        const props = {
            store: mockStore({sharedReducer: {locale: 'ru'} }),
            term: {id: 1, value: 'Тем'},
            indentAfter: true
        };
        expect(mount(<Term {...props} />, {lifecycleExperimental: true})).toMatchSnapshot();
    });

    it('component match expected snapshot for simple term with indent before and after', () => {
        const props = {
            store: mockStore({sharedReducer: {locale: 'ru'} }),
            term: {id: 1, value: 'Тем'},
            indentBefore: true,
            indentAfter: true
        };
        expect(mount(<Term {...props} />, {lifecycleExperimental: true})).toMatchSnapshot();
    });
});
