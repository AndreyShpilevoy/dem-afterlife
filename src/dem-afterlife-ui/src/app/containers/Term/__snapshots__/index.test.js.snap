// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Term component match expected snapshot for not term and untermedPostfix 1`] = `
<Connect(Term)
  doNotTerm={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  untermedPostfix=":"
>
  <Term
    dispatch={[Function]}
    doNotTerm={true}
    locale="ru"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    untermedPostfix=":"
  >
    <span
      style={
        Object {
          "paddingLeft": "",
          "paddingRight": "",
        }
      }
    >
      :
    </span>
  </Term>
</Connect(Term)>
`;

exports[`Term component match expected snapshot for simple term 1`] = `
<Connect(Term)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  term={
    Object {
      "id": 1,
      "value": "Тем",
    }
  }
>
  <Term
    dispatch={[Function]}
    doNotTerm={false}
    locale="ru"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    term={
      Object {
        "id": 1,
        "value": "Тем",
      }
    }
    untermedPostfix=""
  >
    <span
      style={
        Object {
          "paddingLeft": "",
          "paddingRight": "",
        }
      }
    >
      Тем
    </span>
  </Term>
</Connect(Term)>
`;

exports[`Term component match expected snapshot for simple term and untermedPostfix 1`] = `
<Connect(Term)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  term={
    Object {
      "id": 1,
      "value": "Тем",
    }
  }
  untermedPostfix=":"
>
  <Term
    dispatch={[Function]}
    doNotTerm={false}
    locale="ru"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    term={
      Object {
        "id": 1,
        "value": "Тем",
      }
    }
    untermedPostfix=":"
  >
    <span
      style={
        Object {
          "paddingLeft": "",
          "paddingRight": "",
        }
      }
    >
      Тем:
    </span>
  </Term>
</Connect(Term)>
`;

exports[`Term component match expected snapshot for simple term with indent after 1`] = `
<Connect(Term)
  indentAfter={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  term={
    Object {
      "id": 1,
      "value": "Тем",
    }
  }
>
  <Term
    dispatch={[Function]}
    doNotTerm={false}
    indentAfter={true}
    locale="ru"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    term={
      Object {
        "id": 1,
        "value": "Тем",
      }
    }
    untermedPostfix=""
  >
    <span
      style={
        Object {
          "paddingLeft": "",
          "paddingRight": "0.25rem",
        }
      }
    >
      Тем
    </span>
  </Term>
</Connect(Term)>
`;

exports[`Term component match expected snapshot for simple term with indent before and after 1`] = `
<Connect(Term)
  indentAfter={true}
  indentBefore={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  term={
    Object {
      "id": 1,
      "value": "Тем",
    }
  }
>
  <Term
    dispatch={[Function]}
    doNotTerm={false}
    indentAfter={true}
    indentBefore={true}
    locale="ru"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    term={
      Object {
        "id": 1,
        "value": "Тем",
      }
    }
    untermedPostfix=""
  >
    <span
      style={
        Object {
          "paddingLeft": "0.25rem",
          "paddingRight": "0.25rem",
        }
      }
    >
      Тем
    </span>
  </Term>
</Connect(Term)>
`;

exports[`Term component match expected snapshot for simple term with space after 1`] = `
<Connect(Term)
  spaceAfter={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  term={
    Object {
      "id": 1,
      "value": "Тем",
    }
  }
>
  <Term
    dispatch={[Function]}
    doNotTerm={false}
    locale="ru"
    spaceAfter={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    term={
      Object {
        "id": 1,
        "value": "Тем",
      }
    }
    untermedPostfix=""
  >
    <span
      style={
        Object {
          "paddingLeft": "",
          "paddingRight": "",
        }
      }
    >
      Тем 
    </span>
  </Term>
</Connect(Term)>
`;

exports[`Term component match expected snapshot for simple term with space before and after 1`] = `
<Connect(Term)
  spaceAfter={true}
  spaceBefore={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  term={
    Object {
      "id": 1,
      "value": "Тем",
    }
  }
>
  <Term
    dispatch={[Function]}
    doNotTerm={false}
    locale="ru"
    spaceAfter={true}
    spaceBefore={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    term={
      Object {
        "id": 1,
        "value": "Тем",
      }
    }
    untermedPostfix=""
  >
    <span
      style={
        Object {
          "paddingLeft": "",
          "paddingRight": "",
        }
      }
    >
       Тем 
    </span>
  </Term>
</Connect(Term)>
`;

exports[`Term component match expected snapshot for term with indent before 1`] = `
<Connect(Term)
  indentBefore={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  term={
    Object {
      "id": 1,
      "value": "Тем",
    }
  }
>
  <Term
    dispatch={[Function]}
    doNotTerm={false}
    indentBefore={true}
    locale="ru"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    term={
      Object {
        "id": 1,
        "value": "Тем",
      }
    }
    untermedPostfix=""
  >
    <span
      style={
        Object {
          "paddingLeft": "0.25rem",
          "paddingRight": "",
        }
      }
    >
      Тем
    </span>
  </Term>
</Connect(Term)>
`;

exports[`Term component match expected snapshot for term with space before 1`] = `
<Connect(Term)
  spaceBefore={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  term={
    Object {
      "id": 1,
      "value": "Тем",
    }
  }
>
  <Term
    dispatch={[Function]}
    doNotTerm={false}
    locale="ru"
    spaceBefore={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    term={
      Object {
        "id": 1,
        "value": "Тем",
      }
    }
    untermedPostfix=""
  >
    <span
      style={
        Object {
          "paddingLeft": "",
          "paddingRight": "",
        }
      }
    >
       Тем
    </span>
  </Term>
</Connect(Term)>
`;
